yarl/_url.py

```diff
@@ -276,14 +319,14 @@ def __str__(self):
             )
         return urlunsplit(val)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return f"{self.__class__.__name__}('{str(self)}')"
 
-    def __bytes__(self):
+    def __bytes__(self) -> bytes:
         return str(self).encode("ascii")
 
-    def __eq__(self, other):
-        if not type(other) is URL:
+    def __eq__(self, other: Any) -> bool:
```

Comment:
This should be type safe with any object.

```suggestion
    def __eq__(self, other: object) -> bool:
```

Reply:
Same for the other comparison methods.

Reply:
This should be fine to use the `type()` check as its faster and subclassing is forbidden https://github.com/aio-libs/yarl/blob/d4e76015f31d8133ba37d1d2e1a09fbbeb05538f/tests/test_url.py#L18 and blocked in `__init_subclass__`